types/node/index.d.ts

```diff
@@ -32,17 +32,101 @@
 // This needs to be global to avoid TS2403 in case lib.dom.d.ts is present in the same build
 interface Console {
     Console: NodeJS.ConsoleConstructor;
+    /**
+     * A simple assertion test that verifies whether `value` is truthy.
+     * If it is not, an `AssertionError` is thrown.
+     * If provided, the error `message` is formatted using `util.format()` and used as the error message.
+     */
     assert(value: any, message?: string, ...optionalParams: any[]): void;
-    dir(obj: any, options?: NodeJS.InspectOptions): void;
+    /**
+     * When `stdout` is a TTY, calling `console.clear()` will attempt to clear the TTY.
+     * When `stdout` is not a TTY, this method does nothing.
+     */
+    clear(): void;
+    /**
+     * Maintains an internal counter specific to `label` and outputs to `stdout` the number of times `console.count()` has been called with the given `label`.
+     */
+    count(label?: string): void;
+    /**
+     * Resets the internal counter specific to `label`.
+     */
+    countReset(label?: string): void;
+    /**
+     * The `console.debug()` function is an alias for {@link console.log()}.
+     */
     debug(message?: any, ...optionalParams: any[]): void;
+    /**
+     * Uses {@link util.inspect()} on `obj` and prints the resulting string to `stdout`.
+     * This function bypasses any custom `inspect()` function defined on `obj`.
+     */
+    dir(obj: any, options?: NodeJS.InspectOptions): void;
+    /**
+     * The `console.dirxml()` function is an alias for {@link console.log()}.
+     */
+    dirxml(...data: any[]): void;
```

Comment:
Why is there a difference in the type definition of `dirxml` to `log` if one is just an alias of the other?

Reply:
This is based on the Node docs, which say that the method signature is `...data <any>`, and that it "calls console.log() passing it the arguments received".  The [official spec](https://console.spec.whatwg.org/#assert) agrees with Node's definition, even though [the MDN page](https://developer.mozilla.org/en-US/docs/Web/API/Console/dirxml) says it's "non-standard" and only takes exactly one argument.  Of course, logically `foo(...everything: any[])` works identically to `foo(first?: any, ...others: any[])` (from the caller's perspective) so it's kind of academic.

All that to say: I copied it from other similar methods (like `debug`) and probably should have used the original docs from Node,

> This method calls console.log() passing it the arguments received. Please note that this method does not produce any XML formatting.

Reply:
I found following in NodeJS Source:
```Console.prototype.debug = Console.prototype.log;
Console.prototype.info = Console.prototype.log;
Console.prototype.dirxml = Console.prototype.log;```
So seems there is really no difference between them.

But you are right, for the user it's don't care which variant is used here in the typings.